pennylane/ops/arithmetic/adjoint.py

```diff
@@ -0,0 +1,435 @@
+
+    def eigvals(self):
+        # Cannot define ``compute_eigvals`` because Hermitian only defines ``get_eigvals``
+        return [conj(x) for x in self.base.eigvals()]
+
+    # pylint: disable=arguments-differ
+    @staticmethod
+    def compute_diagonalizing_gates(*params, wires, base=None):
+        return base.compute_diagonalizing_gates(*params, wires, **base.hyperparameters)
+
+    # pylint: disable=arguments-renamed
+    @property
+    def has_matrix(self):
+        return self.base.has_matrix
+
+    # pylint: disable=arguments-differ
+    def adjoint(self):
+        return self.base
+
+    @property
+    def _queue_category(self):
+        """Used for sorting objects into their respective lists in `QuantumTape` objects.
+
+        This property is a temporary solution that should not exist long-term and should not be
+        used outside of ``QuantumTape._process_queue``.
+
+        Returns ``_queue_cateogory`` for base operator.
+        """
+        return self.base._queue_category  # pylint: disable=protected-access
+
+    def generator(self):
+        if isinstance(self.base, Operation):  # stand in for being unitary and inverse=adjoint
+            return -1.0 * self.base.generator()
+        return super().generator()
+
+    ## Operation specific properties ##########################################
+
+    @property
+    def basis(self):
+        """str or None: The target operation for controlled gates.
+        target operation. If not ``None``, should take a value of ``"X"``, ``"Y"``,
+        or ``"Z"``.
+
+        For example, ``X`` and ``CNOT`` have ``basis = "X"``, whereas
+        ``ControlledPhaseShift`` and ``RZ`` have ``basis = "Z"``.
+        """
+        return self.base.basis
+
+    @property
+    def control_wires(self):
+        r"""Control wires of the operator.
+
+        For operations that are not controlled,
+        this is an empty ``Wires`` object of length ``0``.
+
+        Returns:
+            Wires: The control wires of the operation.
+        """
+        return self.base.control_wires
+
+    def single_qubit_rot_angles(self):
+        r"""The parameters required to implement a single-qubit gate as an
+        equivalent ``Rot`` gate, up to a global phase.
+
+        Returns:
+            tuple[float, float, float]: A list of values :math:`[\phi, \theta, \omega]`
+            such that :math:`RZ(\omega) RY(\theta) RZ(\phi)` is equivalent to the
+            original operation.
+        """
+        omega, theta, phi = self.base.single_qubit_rot_angles()
+        return [-phi, -theta, -omega]
+
+    @property
+    def grad_method(self):
+        """Gradient computation method.
+
+        * ``'A'``: analytic differentiation using the parameter-shift method.
+        * ``'F'``: finite difference numerical differentiation.
+        * ``None``: the operation may not be differentiated.
+
+        Default is ``'F'``, or ``None`` if the Operation has zero parameters.
+        """
```

Comment:
These are methods from `Operation`, not `Operator`. Since we don't inherit from `Operation`, they don't inherit the docstrings.  Looking into dynamic mixins and the `__new__` method to try and solve this problem.

Reply:
Docstrings are now inherited through `AdjointOperation`.